hazelcast/src/main/java/com/hazelcast/util/function/ToResultSetFunction.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.util.function;
+
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * A function that accepts a JDBC connection to the database, the total
+ * parallelism and processor index as arguments and produces a result set.
+ * This result set should return a part of the whole result set specific to
+ * this processor.
+ */
+@FunctionalInterface
+public interface ToResultSetFunction extends Serializable {
```

Comment:
What is the use-case for this function in Hazelcast? I don't remember anywhere we use JDBC api?

Reply:
True. This entire package was simply copy-pasted from jet as something we might include in IMDG. Perhaps we don't need this interface. On the other hand, I don't see a strong reason why we should introduce any class/interface in this package, unless we start using them in our API. WDYT?

Reply:
I don't have strong opinion... my thinking is we can introduce `j.u.function` counterparts but skip custom ones like `ToResultSetFunction` or `Tri..`, `Quad...` etc.

Reply:
Ok, agree, will do it that way.